src/sentry/models/releaseprojectenvironment.py

```diff
@@ -12,16 +15,60 @@
     region_silo_model,
     sane_repr,
 )
+from sentry.db.models.manager import BaseManager
+from sentry.incidents.utils.types import AlertRuleActivationConditionType
 from sentry.utils import metrics
 from sentry.utils.cache import cache
 
+if TYPE_CHECKING:
+    from sentry.models.environment import Environment
+    from sentry.models.project import Project
+    from sentry.models.release import Release
+    from sentry.snuba.models import QuerySubscription
+
 
 class ReleaseStages(str, Enum):
     ADOPTED = "adopted"
     LOW_ADOPTION = "low_adoption"
     REPLACED = "replaced"
 
 
+class ReleaseProjectEnvironmentManager(BaseManager["ReleaseProjectEnvironment"]):
+    @staticmethod
+    def subscribe_project_to_alert_rule(
+        project: Project, release: Release, environment: Environment, trigger: str
+    ) -> list[QuerySubscription]:
+        """
+        TODO: potentially enable custom query_extra to be passed on ReleaseProject creation (on release/deploy)
+
+        NOTE: import AlertRule model here to avoid circular dependency
+        """
+        from sentry.incidents.models.alert_rule import AlertRule
+
+        query_extra = f"release:{release.version} and environment:{environment.name}"
+        # TODO: parse activator on the client to derive release version / environment name
+        activator = f"release:{release.version} and environment:{environment.name}"
```

Comment:
nit: rather than storing as text blobs, thoughts on building this string in the UI and storing release / env in json? (unclear how large of a refactor this ask is though).

Also, curious why we need to replicate this data? Should the activator only store activation reasons, and we remove activation info from query_extra? My understanding is that query_extra is meant to be meta data about the query itself stored in text. The activator is meant to be the release / deploy / environment information that triggered the alert on activated alerts.

Do we need to do anything to anticipate deploy version info here?

Reply:
Yea - query_extra is actually being tacked on to the query.

`activator` is simply something we can provide to the client to help better understand what triggered the monitor.
For now i mirrored it with the query_extra, btu we could make this any string 🤷 

Deploy version isn't valuable and we do not surface it anywhere else in the UI either :( 
'deploy's are simply releases on certain environments, and that's how we surface it elsewhere so stuck wtih that

Reply:
This trigger isn't being generated from any UI, so we cannot build anything in the UI / have nothing to store in JSON

Reply:
I was more thinking that the format could be store as json rather than a string - incase we want to parse it later or share it via an API. it's just a nice to have that might make the future a little easier. Basically, just doing: `json.dumps({ release: release.version, environment: environent.name })` then we can parse it with `json.loads()` on fetch. Just thinking it might be nicer to interact with this as a dictionary than as a string incase we need the release / env name later, but isn't a blocker.

The thing i was thinking with deploys is that in the future we might want to have an activator be a specific deploy, but def don't need to get hung-up on it for the mvp either - so :punt: